participant/batch/refreshS3Urls.ts

```diff
@@ -0,0 +1,106 @@
+// A batch script to update s3 presigned urls saved to the database.
+// Intended to be run in a lambda.
+import {
+  listExpiringDocuments,
+  updateDocumentS3Url,
+} from "app/utils/db.server";
+import { getURLFromS3 } from "app/utils/s3.server";
+import logger from "app/utils/logging.server";
+
+// Any interactions with Prisma will be async
+// eslint-disable-next-line @typescript-eslint/require-await
+export async function main() {
+  logger.warn(
+    {
+      location: "refreshS3Urls",
+      type: "main.begin",
+    },
+    "Beginning s3 document upload url refresh"
+  );
+
+  const documentsToRefresh = await listExpiringDocuments();
+  if (documentsToRefresh !== undefined && documentsToRefresh.length > 0) {
+    logger.warn(
+      {
+        location: "refreshS3Urls",
+        type: "main.found_expiring_documents",
+      },
+      `Refreshing ${documentsToRefresh.length} document urls`
+    );
+
+    for (const document of documentsToRefresh) {
+      let newS3Url;
+      try {
+        newS3Url = await getURLFromS3(document.s3Key);
+        logger.info(
+          {
+            location: "refreshS3Urls",
+            type: "main.get_url",
+            submissionId: document.submissionId,
+            filename: document.originalFilename,
+            lastUpdatedAt: document.updatedAt,
+          },
+          `New document url: ${newS3Url}`
+        );
+      } catch (error) {
+        logger.error(
+          {
+            location: "refreshS3Urls",
+            type: "main.get_url",
+            submissionId: document.submissionId,
+            filename: document.originalFilename,
+            lastUpdatedAt: document.updatedAt,
+          },
+          `Error encountered trying to get new URL: ${error}`
+        );
```

Comment:
Why a `let` and a test instead of just continuing to the next for at the `catch` above?
```suggestion
        );
        continue;
```

Reply:
I need to access `newS3Url` on line 57. I could put it all into one `try` block, but I wanted to capture different log messages